stream-product/product-ingestion-saga/src/main/java/com/backbase/stream/product/BatchProductIngestionSaga.java

```diff
@@ -198,7 +198,8 @@ protected Mono<BatchProductGroupTask> upsertArrangementsBatch(BatchProductGroupT
                 .collectList()
                 .thenReturn(batchProductGroupTask)
                 .flatMap(task -> {
-                    if (task.getIngestionMode().equals(BatchProductGroupTask.IngestionMode.REPLACE)) {
+                    if (task.isDeleteArrangementsInReplaceMode() &&
```

Comment:
in my perspective a new ingestion mode would make more sense here, there is no `else` condition meaning that if this flag you included is set to false there will be no replacement at all

Reply:
That's the point. Technically, this entire block of code should be removed because arrangement deletion is problematic based on entitlements for one user. However, to avoid introducing breaking change to ALL customers using this version of Streams without a warning this flag is added. It's a toggle so the else is that the enclosed code doesn't execute. Those that don't want to delete will set the flag to false during BatchProductGroupTask creation. Others won't be affected at all, i.e. no changes for them.
The IngestionMode is still REPLACE and is used specifically for account group and permission updates (AccessGroupService).

Reply:
how by adding a new ingestion mode will be a breaking change?